### PR TITLE
Added oAuth for GitHub

### PR DESCRIPTION
**Pull request by RasmusAChr**
# Features
## Added GitHub OAuth Login Integration
- Integrated GitHub OAuth authentication as a login method.
- Added the `AspNet.Security.OAuth.GitHub` package.
- Configured GitHub OAuth authentication in the builder pipeline and configured session management including client ID, client secret, and callback path.
- Enabled session management to support GitHub login state persistence.

## Link to 'Manage Account' page
- Added a separate link to the "Manage Account" page in the navigation bar for easier access to account settings. This page was always accessible but previously not linked directly in the navigation bar, as Microsoft hides some pages by default. We have just created a direct link to enhance the user navigation.

# How to set up secrets
We are now using secret keys for github oAuth both locally and on Azure. This means that if you have not initialized the secret key locally on your machine, the program **will not run!** To solve this I have this step by step guide to set up a secret key below:
1. In our organization, go to `Settings` and scroll down to `Developer Settings`
2. Click on `Developer Settings` and then `OAuth Apps`.
3. Click on `Chirp Locally`. Do not click on `Chirp` since that is used for Azure and not locally.
4. Click on `Generate a new client secret`.
5. Save the `client id` and `client secret`.
6. Now open the terminal and write:
    1. `dotnet user-secrets init`.
    2. `dotnet user-secrets set "authentication_github_clientId" "<YOUR_CLIENTID>"` without the `<>`.
    3. `dotnet user-secrets set "authentication_github_clientSecret" "<YOUR_CLIENTSECRET>"` without the `<>`.
7. Now you will be able to run the program using `dotnet watch` in the `Chirp.Web` project.
DISCLAIMER: You will not be able to use the github login without being on `dotnet watch`, because the Github OAuth Application is tied up to the port that `dotnet watch` makes when running.
DISCLAIMER part 2: We only have 5 client secrets available in Github Settings. Hence it would be optimal for all of us to use just use the same client secret. However the secret can only be viewed once in Github, so this has to be coordinated a day we are all together.

# Known Issues
The defaultConnection string for the local database i appsettings.json doesnt work locally on Linux machines. An [issue](https://github.com/ITU-BDSA2024-GROUP7/Chirp/issues/99#issue-2621022773) has been made to give focus to this problem

### CODE REVIEW

**Review by Nikolaj787 — APPROVED** [review]
Great changes to the program. And yes let's coordinate so that we all can just use one secret and not use all 5 available secrets

File: src/Chirp.Infrastructure/Data/Cheep.db
Comment (DragonFlyersx) [inline src/Chirp.Infrastructure/Data/Cheep.db:1]: We need to add the default accounts for Helge and Adrian

**Review by DragonFlyersx — APPROVED** [review]
Overall The Code looks like a good implementation of Oauth for Github,

**Review by niko391a — APPROVED** [review]
Clean implementation, great that it could be done with so few lines :)

**Review by Tipskind — APPROVED** [review]
builder.Services.AddAuthentication(), adding secret looks good. Additional authentication looks good.

**Pull request merged**